### PR TITLE
Convert import hints to ghc-lib-parser

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR rewrites the import module hints in terms of the GHC parse tree.

### CODE REVIEW

**Comment by shayne-fletcher-da** [comment]
Relates to issue https://github.com/ndmitchell/hlint/issues/645.

File: src/GHC/Util.hs [hunk 1]
```diff
@@ -259,3 +264,26 @@ declName (SigD _ (ClassOpSig _ _ (x:_) _)) = occNameString $ occName $ unLoc x
...
+
+-- | Compare two 'Maybe (Located a)' values for equality modulo
+-- locations.
+eqMaybe:: Eq a => Maybe (Located a) -> Maybe (Located a) -> Bool
+eqMaybe (Just x) (Just y) = x =~= y
+eqMaybe Nothing Nothing = True
+eqMaybe _ _ = False
+
+noloc :: e -> Located e
+noloc = noLoc
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply src/GHC/Util.hs]: Conflict with HSE. They'd have to spelled `GHC.noLoc`.

**Comment by shayne-fletcher-da** [comment]
> Can you fix the merge conflicts and resubmit?
Rebased

**Comment by shayne-fletcher-da** [comment]
The Travis test failures seem to relate to https://github.com/ndmitchell/hlint/issues/712.

**Comment by ndmitchell** [comment]
Thanks, I hadn't quite hit all the places UNIPLATE_VERBOSE was set. Have merged and done that. Thanks for the code!

**Comment by googleson78** [comment]
Thanks for adding all the types on `idea` functions!

**Pull request merged**